tools/build_pak_files.py

```diff
@@ -0,0 +1,135 @@
+    parser.add_argument("--platforms", nargs="*", required=True)
+    parser.add_argument("--output_dir", required=True)
+    args = parser.parse_args()
+    
+    assert os.path.exists(args.unreal_engine_dir)
+    if sys.platform == "win32":
+        unreal_editor_bin = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Win64", "UE4Editor.exe"))
+        unreal_pak_bin    = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Win64", "UnrealPak.exe"))
+    elif sys.platform == "darwin":
+        unreal_editor_bin = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Mac", "UE4Editor"))
+        unreal_pak_bin    = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Mac", "UnrealPak"))
+    elif sys.platform == "linux":
+        unreal_editor_bin = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Linux", "UE4Editor.sh"))
+        unreal_pak_bin    = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Linux", "UnrealPak.sh"))
+
+    unreal_project_dir         = os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "cpp", "unreal_projects", "SpearSim"))
+    uproject                   = os.path.join(unreal_project_dir, "SpearSim.uproject")
+    unreal_project_content_dir = os.path.join(unreal_project_dir, "Content")
+
+    assert os.path.exists(args.contents_dir)
+    shared_dir = os.path.realpath(os.path.join(args.contents_dir, "Shared"))
+    scenes_dir = os.path.realpath(os.path.join(args.contents_dir, "Scenes"))
+
+    assert os.path.exists(shared_dir)
+    unreal_project_content_shared_dir = os.path.join(unreal_project_content_dir, "Shared")
+    if not spear.path_exists(unreal_project_content_shared_dir):
+        print(f"[SPEAR | build_pak_files.py] Creating symlink: {unreal_project_content_shared_dir} -> {shared_dir}")
+        os.symlink(shared_dir, unreal_project_content_shared_dir)
+
+    assert os.path.exists(scenes_dir)
+    scene_content_dirs = [ os.path.realpath(os.path.join(scenes_dir, x)) for x in os.listdir(scenes_dir) if x.startswith("kujiale") ]
+    assert len(scene_content_dirs) > 0
+
+    for scene_content_dir in scene_content_dirs:
+
+        scene_name = os.path.basename(scene_content_dir)
+        unreal_project_content_scene_dir = os.path.join(unreal_project_content_dir, "Scenes", scene_name)
+
+        # remove existing scene dir
+        if spear.path_exists(unreal_project_content_scene_dir):
+            print(f"[SPEAR | build_pak_files.py] File or directory or symlink exists, removing: {unreal_project_content_scene_dir}")
+            spear.remove_path(unreal_project_content_scene_dir)
+
+        # create symlink
+        print(f"[SPEAR | build_pak_files.py] Creating symlink: {unreal_project_content_scene_dir} -> {scene_content_dir}")
+        os.symlink(scene_content_dir, unreal_project_content_scene_dir)
+        
+        for platform in args.platforms:
+
+            cmd = [
```

Comment:
Where do all these arguments come from?

Reply:
I'm not sure about all the params. I could only find some [here](https://docs.unrealengine.com/4.26/en-US/SharingAndReleasing/Deployment/Cooking/). @rwang15, do you have more information on this?